src/rasberry_perception/interfaces/scene_analyzer_server.py

```diff
@@ -0,0 +1,171 @@
+
+    
+
+"""
+Repurposed from scene_analyser from FastPick
+"""
+
+class MasksPredictor:
+
+    def __init__(self, model_file, config_file, metadata_file, num_classes  = 3 , accuracy = 0.01):
+        
+        self.counter = 1
+        self.accuracy = accuracy 
+        self.scale=1.0
+
+        self.metadata=self.get_metadata(metadata_file)
+        self.cfg = self.init_config(model_file, config_file, num_classes)
+
+        print ( self.metadata)
+        print ( self.cfg.DATASETS.TRAIN[0] )
+        try:
+            self.predictor=DefaultPredictor(self.cfg)
+        except Exception as e:
+            logging.error(e)
+            print(e)
+            raise Exception(e)
+
+    def init_config(self, model_file, config_file, num_classes):
+        cfg = get_cfg()
+        try:
+            cfg.merge_from_file(model_zoo.get_config_file(config_file))
+        except Exception as e:
+            logging.error(e)
+            print(e)
+            raise Exception(e)
+
+        cfg.MODEL.WEIGHTS = model_file #os.path.join(model_file)
+        cfg.MODEL.ROI_HEADS.NUM_CLASSES = num_classes  # 1.strawberry, 2. Canopy, 3. Rigid Structure
+        return cfg
+
+    def get_metadata(self,metadata_file):
+
+        #metadata file has name of classes, it is created to avoid having custom dataset and taking definitions
+        # from annotations, instead. It has structure of MetaDataCatlog output of detectron2
+        try:
+            file = open(metadata_file, 'rb')
+        except Exception as e:
+            logging.error(e)
+            print(e)
+            raise Exception(e)
+
+        data = pickle.load(file)
+        file.close()
+        return data
+
+    def get_predictions(self, ros_rgb_image ):
+        
+
+        np_rgb_image = ros_numpy.numpify( ros_rgb_image )
+        outputs = self.predictor( np_rgb_image )
+
+        v = Visualizer(np_rgb_image[:, :, ::-1], self.metadata , scale=1.0)
+        out = v.draw_instance_predictions(outputs["instances"].to("cpu"))
+
+        pred_classes = outputs["instances"].get_fields()["pred_classes"].to("cpu").numpy()
+        scores = outputs["instances"].get_fields()["scores"].to("cpu").numpy()
+        pred_masks = outputs["instances"].get_fields()["pred_masks"].to("cpu").numpy()
+        pred_boxes = outputs["instances"].get_fields()["pred_boxes"].to("cpu").tensor.numpy()
+
+        
+        prediction_output = {
+            "pred_classes" : pred_classes,
+            "scores" : scores,
+            "pred_masks" : pred_masks,
+            "pred_boxes" : pred_boxes,
+        }            
+ 
+
+        output_image = out.get_image()[:, :, ::-1].astype(np.uint8)
+        output_filepath = "/home/arshad/Desktop/outputs/output_{}.png".format( self.counter )
+        cv2.imwrite( output_filepath , np.concatenate( (cv2.cvtColor(np_rgb_image, cv2.COLOR_RGB2BGR), cv2.cvtColor(output_image.copy(), cv2.COLOR_RGB2BGR) ), axis=0 ) ) 
```

Comment:
Why are you writing to desktop here? This will slow down people's IO and fail on machines that don't have your user name. Remove specific debugging code If you want a image writing script either add it as an argument or create a separate ros node to save images to disk (these already exist)